lightning-network-book.tex

```diff
@@ -240,36 +240,36 @@
 \newglossaryentry{lnd}{name={lnd},
-  description={Implementation of the Lightning Network Protocol by the San Francisco based company Lightning Labs.
+  description={Implementation of the Lightning Network Protocol by the San Francisco based company Lightning Labs (\url{https://lightning.engineering}). Source code is at \url{https://github.com/lightningnetwork/lnd}.
 It is written in Go.}}
 \newglossaryentry{eclair}{name={Eclair},
-  description={Implementation of the Lightning Network Protocol by the Paris based company ACINQ. It is written in Scala.}}
+  description={Implementation of the Lightning Network Protocol by the Paris based company ACINQ (\url{https://acinq.co}). Source code is at \url{https://github.com/ACINQ/eclair}. It is written in Scala.}}
 \newglossaryentry{clightning}{name={c-lightning},
-  description={Implementation of the Lightning Network Protocol by the Victoria based Blockstream. It is written in C.}}
+  description={Implementation of the Lightning Network Protocol by the Victoria based Blockstream(\url{https://blockstream.com}). Source code is at \url{https://github.com/ElementsProject/lightning}. It is written in C.}}
 \newglossaryentry{shachain}{name={Sha-chain},
   description={}}
 \newglossaryentry{hdwallet}{name={HD Wallet},
   description={}}
 \newglossaryentry{bip}{name={BIP},
-  description={}}
+  description={BIP is short for Bitcoin Improvement Proposals. It is a mechanism by which proposals for the improvement of the Bitcoin network can be made. Submitted proposals are discussed heavily and the best one implemented. See \url{https://github.com/bitcoin/bips}.}}
 \newglossaryentry{bitcoin}{name={Bitcoin},
-  description={}}
+  description={Bitcoin is an experimental digital currency that enables payments to anyone, anywhere in the world. Bitcoin uses peer-to-peer technology to operate with no central authority: managing transactions and issuing money are carried out collectively by the network. See \url{https://www.bitcoin.org} and \url{https://github.com/bitcoin/bitcoin}.}}
 \newglossaryentry{ecash}{name={Electronic Cash},
   description={}}
 \newglossaryentry{ecdsa}{name={ECDSA},
   description={}}
 \newglossaryentry{bech32}{name={bech32},
-  description={}}
+  description={Bech32 is a segwit address format specified by BIP 0173. This address format is also known as "bc1 addresses" because all Bech32 addresses start with "bc" and currently with "bc1". See \url{https://en.bitcoin.it/wiki/Bech32}.}}
 \newglossaryentry{encoding}{name={Encoding},
-  description={}}
+  description={Encoding is the act of translating information from one form to another following specified rules. Text can be encoded as morse signals. Songs can be encoded as MP3 files. The opposite transformation is called "decoding". An MP3 song is decoded as audio.}}
 \newglossaryentry{input}{name={Input Script},
-  description={}}
+  description={The Input Script is a script to prove that you own some bitcoins.}}
```

Comment:
what about multisig scripts? what about time lock scripts and smart contracts? wouldn't it make more sense to define this in compination with the output script?

Reply:
```suggestion
  description={In the absolute most basic form the input script is used to prove that you own some bitcoins. Look at the glossary entry "Bitcoin Script" for more details about what scripts can do and what role they play.}}
```